### PR TITLE
chore: Bump rubocop 1.41.1

### PR DESCRIPTION
**Pull request by arielvalentin**
Bumps rubocop to 1.41.1 and explicitly disables rules that were introduced since the last upgrade. 

I will submit follow-up PRs that enables these newer rules so that we may review them in isolation and discuss the merit of adopting them.

### CODE REVIEW

File: resource_detectors/lib/opentelemetry/resource/detectors/google_cloud_platform.rb [hunk 1]
```diff
@@ -35,7 +35,7 @@ def detect # rubocop:disable Metrics/CyclomaticComplexity, Metrics/MethodLength,
             resource_attributes[OpenTelemetry::SemanticConventions::Resource::K8S_POD_NAME] = ENV['HOSTNAME'] || safe_gethostname
             resource_attributes[OpenTelemetry::SemanticConventions::Resource::K8S_NODE_NAME] = gcp_env.lookup_metadata('instance', 'hostname')
 
-            resource_attributes[OpenTelemetry::SemanticConventions::Resource::CONTAINER_NAME] = ENV['CONTAINER_NAME']
+            resource_attributes[OpenTelemetry::SemanticConventions::Resource::CONTAINER_NAME] = ENV['CONTAINER_NAME'] if ENV['CONTAINER_NAME']
```
Comment (plantfansam) [inline resource_detectors/lib/opentelemetry/resource/detectors/google_cloud_platform.rb:38]: Any reason not to do `# rubocop:disable Style/FetchEnvVar` here, as you did in the `cloud_platform_test`?

**Review by plantfansam — APPROVED** [review]
LGTM - thanks for taking the time to break this up into multiple PRs!

File: resource_detectors/lib/opentelemetry/resource/detectors/google_cloud_platform.rb (hunk 1 above)
Comment (arielvalentin) [reply resource_detectors/lib/opentelemetry/resource/detectors/google_cloud_platform.rb:38]: very sorry those changes snuck in there. I have a separate PR for each linter rule that I am loading up

**Review by plantfansam — APPROVED** [review]

File: .instrumentation_generator/templates/gemspec.tt [hunk 2]
```diff
@@ -34,7 +34,7 @@ Gem::Specification.new do |spec|
   spec.add_development_dependency 'opentelemetry-sdk', '~> 1.0'
   spec.add_development_dependency 'opentelemetry-test-helpers'
   spec.add_development_dependency 'rake', '~> 12.3.3'
-  spec.add_development_dependency 'rubocop', '~> 1.3.0'
+  spec.add_development_dependency 'rubocop', '~> 1.40.0'
```
Comment (arielvalentin) [inline .instrumentation_generator/templates/gemspec.tt:37]:
```suggestion
  spec.add_development_dependency 'rubocop', '~> 1.41.0'
```

**Review by fbogsany — APPROVED** [review]

**Pull request merged**